opentracing_shim/src/main/java/io/opentelemetry/opentracingshim/SpanContextShim.java

```diff
@@ -16,20 +16,51 @@
 
 package io.opentelemetry.opentracingshim;
 
+import io.opentelemetry.distributedcontext.DistributedContext;
+import io.opentelemetry.distributedcontext.EmptyDistributedContext;
+import io.opentelemetry.distributedcontext.Entry;
+import io.opentelemetry.distributedcontext.EntryKey;
+import io.opentelemetry.distributedcontext.EntryMetadata;
+import io.opentelemetry.distributedcontext.EntryValue;
 import io.opentracing.SpanContext;
+import java.util.Iterator;
 import java.util.Map;
 
-final class SpanContextShim implements SpanContext {
-  io.opentelemetry.trace.SpanContext context;
+final class SpanContextShim extends BaseShimObject implements SpanContext {
+  static final EntryMetadata DEFAULT_ENTRY_METADATA =
+      EntryMetadata.create(EntryMetadata.EntryTtl.UNLIMITED_PROPAGATION);
 
-  public SpanContextShim(io.opentelemetry.trace.SpanContext context) {
+  private final io.opentelemetry.trace.SpanContext context;
+  private final DistributedContext distContext;
+
+  public SpanContextShim(TelemetryInfo telemetryInfo, io.opentelemetry.trace.SpanContext context) {
+    this(telemetryInfo, context, EmptyDistributedContext.getInstance());
+  }
+
+  public SpanContextShim(
+      TelemetryInfo telemetryInfo,
+      io.opentelemetry.trace.SpanContext context,
+      DistributedContext distContext) {
+    super(telemetryInfo);
     this.context = context;
+    this.distContext = distContext;
+  }
+
+  SpanContextShim newWithKeyValue(String key, String value) {
+    DistributedContext.Builder builder = contextManager().contextBuilder().setParent(distContext);
+    builder.put(EntryKey.create(key), EntryValue.create(value), DEFAULT_ENTRY_METADATA);
+
+    return new SpanContextShim(telemetryInfo(), context, builder.build());
```

Comment:
So this creates a new D-Context, but does not activate it?

Reply:
by activating I mean `Scope withContext(DistributedContext distContext);`

Reply:
Correct - I'm not doing **any** D-Context activation at all (at this time). We can revisit this later on, I feel...

(Although it *feels* like it will be complicated, as we would need to activate/reactivate the D-Context every time `io.opentracing.Span.setBaggageItem()` is called, and take into account the case when the activation stack has changed...)

Reply:
I can see this causing certain issues (e.g. populating MDC for logging in listeners of context changes), but I don't see how we can solve that given the fundamental inversion of context and span relationship in OTel.